src/cmds/openapi/reduce.ts

```diff
@@ -0,0 +1,162 @@
+import type { CommandOptions } from '../../lib/baseCommand';
+
+import fs from 'fs';
+
+import chalk from 'chalk';
+import jsonpath from 'jsonpath';
+import oasReducer from 'oas/dist/lib/reducer';
+
+import Command, { CommandCategories } from '../../lib/baseCommand';
+import prepareOas from '../../lib/prepareOas';
+import promptTerminal from '../../lib/promptWrapper';
+
+export type Options = {
+  spec?: string;
+  workingDirectory?: string;
+};
+
+export default class OpenAPIReduceCommand extends Command {
+  constructor() {
+    super();
+
+    this.command = 'openapi:reduce';
+    this.usage = 'openapi:reduce [file] [options]';
+    this.description = 'Reduce an OpenAPI definition into a smaller subset.';
+    this.cmdCategory = CommandCategories.APIS;
+    this.position = 3;
+
+    this.hiddenArgs = ['spec'];
+    this.args = [
+      {
+        name: 'spec',
+        type: String,
+        defaultOption: true,
+      },
+      {
+        name: 'workingDirectory',
+        type: String,
+        description: 'Working directory (for usage with relative external references)',
+      },
+    ];
+  }
+
+  async run(opts: CommandOptions<Options>) {
+    super.run(opts);
+
+    const { spec, workingDirectory } = opts;
+
+    if (workingDirectory) {
+      process.chdir(workingDirectory);
+    }
+
+    const { bundledSpec, specPath, specType } = await prepareOas(spec, 'openapi:reduce');
+    const parsedBundledSpec = JSON.parse(bundledSpec);
+
+    if (specType !== 'OpenAPI') {
+      throw new Error('Sorry, this reducer feature in rdme only supports OpenAPI 3.0+ definitions.');
+    }
+
+    const promptResults = await promptTerminal([
+      {
+        type: 'select',
+        name: 'reduceBy',
+        message: 'Would you like to reduce by paths or tags?',
+        initial: 0,
```

Comment:
nit but I don't _think_ you need a default here, I believe the `select` prompt just selects the first one by default:
```suggestion
```